SimpleNotePad.java

```diff
@@ -0,0 +1,213 @@
+			textPane.read(input, "Opening...");    
+		}    			
+		catch (IOException ex) {
+			JOptionPane.showMessageDialog(null,
+					"Open error" + ex, "Opening error",
+					JOptionPane.ERROR_MESSAGE);
+		}
+	}
+
+	//called by open() to update recent files list
+	private void updateRecentFiles(File file) {
+		if(recentFiles.contains(file)) { //removes file from List if it was opened recently to avoid duplicates
+			recentFiles.remove(file);
+		}
+		if(recentFiles.size() == 5) { //removes last item in list if size == 5
+			recentFiles.removeLast();
+		}
+		recentFiles.addFirst(file); //adds file being opened to top of List
+
+		recentMenu.removeAll(); //clears recent menu 
+
+		//repopulates recent menu using updated recent file order
+		for(File f: recentFiles) {
+			JMenuItem recent = new JMenuItem(f.getName());
+			recent.addActionListener(this::action_openRecent);
+			recent.setActionCommand(f.getAbsolutePath());
+			recentMenu.add(recent);
+		} 	
+	}
+
+	//new feature: opens file from system dialog
+	private void action_openFile(ActionEvent e) {
+		JFileChooser fc = new JFileChooser();
+		int returnVal = fc.showOpenDialog(null);
+		if (returnVal == JFileChooser.APPROVE_OPTION) {
+			open(fc.getSelectedFile());
+		}
+	}
+
+	//copy method extracted from actionPerformed
+	private void action_copy(ActionEvent e) {
+		textPane.copy();
+	}
+
+	//paste method extracted from actionPerformed
+	private void action_paste(ActionEvent e) {
+		StyledDocument doc = textPane.getStyledDocument();
+		Position position = doc.getEndPosition();
+		System.out.println("offset"+position.getOffset());
+		textPane.paste();
+	}
+
+	//printFile method extracted from actionPerformed
+	private void action_printFile(ActionEvent e) {
+		try{
+			PrinterJob pjob = PrinterJob.getPrinterJob();
+			pjob.setJobName("Sample Command Pattern");
+			pjob.setCopies(1);
+			pjob.setPrintable(new Printable() {
+				public int print(Graphics pg, PageFormat pf, int pageNum) {
+					if (pageNum>0)
+						return Printable.NO_SUCH_PAGE;
+					pg.drawString(textPane.getText(), 500, 500);
+					paint(pg);
+					return Printable.PAGE_EXISTS;
+				}
+			});
+
+			if (pjob.printDialog() == false)
+				return;
+			pjob.print();
+		} catch (PrinterException pe) {
+			JOptionPane.showMessageDialog(null,
+					"Printer error" + pe, "Printing error",
+					JOptionPane.ERROR_MESSAGE);
+		}
+	}
+
+	//saveFile method extracted from actionPerformed
+	private void action_saveFile(ActionEvent e) {
+		File fileToWrite = null;
+		JFileChooser fc = new JFileChooser();
+		int returnVal = fc.showSaveDialog(null);
+		if (returnVal == JFileChooser.APPROVE_OPTION)
+			fileToWrite = fc.getSelectedFile();
+		try {
+			PrintWriter out = new PrintWriter(new FileWriter(fileToWrite));
+			out.println(textPane.getText());
+			JOptionPane.showMessageDialog(null, "File is saved successfully...");
+			out.close();
+			updateRecentFiles(fileToWrite);
+		}    
+		//implement saving error
```

Comment:
Great idea in making a saving error!  I didn't think of creating a MessageDialog to show the error message after the try catch.